sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/patch/PatchOperation.java

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.patch;
+
+import com.azure.cosmos.patch.implementation.PatchOperationType;
+
+public abstract class PatchOperation<TResource> {
```

Comment:
we are following a slightly different approach for Patch compared to TransactionalBatch PR.

in TransactionalBatch the operations could be instantiated using TransactionalBatch as a builder.
whereas here Operations have their known way of instantiation.

Any reason we can't do the same?

This is for discussion only for now. I am not requesting we should change this.

Reply:
Yeah. As discussed/finalized in meeting, we have moved to CosmosPatch as a builder approach.